### PR TITLE
PC-policy-team-build-once || AWS NAT Gateways should be utilized for the default route

### PR DESCRIPTION
**Pull request by C-Nandhini**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

**PC Policy ID** - bbfd1fec-c777-4265-a307-fbca4a5912e7

**Compliance Standard** - APRA (CPS 234) Information Security-CPS234-23, APRA (CPS 234) Information Security-CPS234-27, APRA (CPS 234) Information Security-CPS234-34, Brazilian Data Protection Law (LGPD)-Article 49, CSA CCM v.4.0.1-AIS-01, CSA CCM v.4.0.1-AIS-02, CSA CCM v.4.0.1-AIS-04, CSA CCM v.4.0.1-CCC-01, CSA CCM v.4.0.1-GRC-03, CSA CCM v.4.0.1-IVS-04, CSA CCM v.4.0.1-UEM-06, CyberSecurity Law of the People's Republic of China-Article 21, Cybersecurity Maturity Model Certification (CMMC) v.1.02-CM.2.062, HITRUST v.9.4.2-Control Reference:01.d, HITRUST v.9.4.2-Control Reference:01.r, ISO/IEC 27002:2013-12.1.2, ISO/IEC 27002:2013-14.1.1, ISO/IEC 27002:2013-14.1.2, ISO/IEC 27002:2013-14.2.1, ISO/IEC 27002:2013-14.2.2, ISO/IEC 27017:2015-12.1.2, ISO/IEC 27017:2015-14.1.1, ISO/IEC 27017:2015-14.1.2, ISO/IEC 27017:2015-14.2.1, ISO/IEC 27017:2015-14.2.5, ISO/IEC 27017:2015-5.1.1, ISO/IEC 27018:2019-12.1.2, MAS TRM 2021-7.2.1, MAS TRM 2021-7.2.2, MLPS 2.0-8.1.3.2, Marcos Soares test standard-Article 49, NIST CSF-PR.IP-1, NIST SP 800-171 Revision 2-3.4.2, NIST SP 800-172-3.13.4e, PCI DSS v3.2.1-2.1

**Remediation Steps:**

**To create a NAT gateway:**
1. Sign into the AWS console
2. In the console, select the specific region from region drop down on the top right corner, for which the alert is generated
3. Navigate to VPC Dashboard
4. In the navigation pane, choose 'NAT Gateways'
5. Click on 'Create NAT Gateway', Specify the subnet in which to create the NAT gateway, and select the allocation ID of an Elastic IP address to associate with the NAT gateway. When you're done, Click on 'Create a NAT Gateway'. The NAT gateway displays in the console. After a few moments, its status changes to Available, after which it's ready for you to use.

**To update Route Table:**
After you've created your NAT gateway, you must update your route tables for your private subnets to point internet traffic to the NAT gateway. We use the most specific route that matches the traffic to determine how to route the traffic.
1. Sign into the AWS console
2. In the console, select the specific region from region drop down on the top right corner, for which the alert is generated
3. Navigate to VPC Dashboard
4. In the navigation pane, choose 'Route Tables'
5. Select the reported route table associated with your private subnet 
6. Choose 'Routes' and Click on 'Edit routes'
7. Replace the current route that points to the NAT instance with a route to the NAT gateway
8. Click on 'Save routes'

### CODE REVIEW

**Comment by C-Nandhini** [comment]
@nimrodkor - Please review and provide your comments.

File: checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  name: "AWS NAT Gateways should be utilized for the default route"
+  id: "CKV2_AWS_35"
+  guidelines: "This policy identifies Route Tables which have NAT instances for the default route instead of NAT gateways. It is recommended to use NAT gateways as the AWS managed NAT Gateway provides a scalable and resilient method for allowing outbound internet traffic from your private VPC subnets."
+  category: "networking"
+  severity: "medium"
+scope:
+  provider: "aws"
+definition:
+    or:
+        - or:
```
Comment (nimrodkor) [inline checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml:12]: @C-Nandhini What's the point of having a nested `or`?

**Review by nimrodkor — COMMENTED** [review]
@C-Nandhini Please resolve the merge conflict 🙏

File: checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml (hunk 1 above)
Comment (C-Nandhini) [reply checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml:12]: We are checking two different resource types route_table and route.
The policy should alert if either one of the resources is incorrectly configured.
Also without the nested OR I got a value error in the editor during testing.
Please advise if there is a different way to achieve the logic.
Screenshot 2022-01-23 at 7 44 48 PM

Comment (nimrodkor) [reply checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml:12]: @C-Nandhini I do think a flat `or` will achieve the same results, if you want to test it

Comment (C-Nandhini) [reply checkov/terraform/checks/graph_checks/aws/AWSNATGatewaysshouldbeutilized.yaml]: @nimrodkor - Yeah that worked !!!
Thanks for the suggestion, I was of the impression to use OR if the resource type is different.
Surely will correct in future.

**Review by nimrodkor — APPROVED** [review]

**Comment by nimrodkor** [comment]
@C-Nandhini It seems the new UT you wrote fails:
`E   AssertionError: 4 != 5 : mismatch in number of results in passed, expected: 4, got: 5`

**Comment by nimrodkor** [comment]
@C-Nandhini Your UT still does not pass

**Comment by C-Nandhini** [comment]
@nimrodkor - The assertion error is fixed in the latest and the test cases passed in my local testing.
So Please review the changes.

**Comment by nimrodkor** [comment]
Hey @C-Nandhini !

Two things:
1. The pipenv GH action broke, please merge from master so the UTs will run.
2. `severity` is not a supported attribute per the [YAML policy docs](https://www.checkov.io/3.Custom%20Policies/YAML%20Custom%20Policies.html). Please remove it

**Comment by C-Nandhini** [comment]
@nimrodkor - Merged changes from master and removed severity as advised.
Please review.

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**